pillar/vars.sls

```diff
@@ -5,3 +5,6 @@ dashboard:      'dashboard'
 
 # The external FQDN for velum
 dashboard_external_fqdn: ''
+
+# List of the nodes which have been removed from CaaSP
+caasp_removed_nodes: [ ]
```

Comment:
I wouldn't use the pillar for this, as the pillar is meant more as a static place to save information.

Instead, when we from the UI mark some nodes as to be removed, we could write a grain on all those machines (`node_to_be_removed: True`), for example, and then call to the `orch.remove-nodes` orchestration from the UI using `salt-api`.

Reply:
I think the pillar is fine because it can be used as a variable for the orchestration (so you have a _parametrized orchestration_) while a grain must be set in a previous step (and then you have to control some other things).

Reply:
Oh, I would be perfectly fine with that (parametrized orchestration), basically doing a `salt-run state.orchestrate orch.remove_nodes pillar={remove_nodes:["a", "b"]}` but from the `salt-api`.

In that case I'd remove this bit from here as this pillar variable is of no interest to the rest of the orchestrations and we'll always call to `orch.remove_nodes` orchestration with the right parameters.

Reply:
I agree with @ereslibre: the pillar value is of no interest to _users_. The convention in Salt's docs is to use a `target` as the parameter for this kind of orchestrations. I would also restrict the removal to only one target (not a list), as users would probably not want to remove many nodes at the same time (and that would require many commplicated checks and maneuvers on our side).